spec/locale.html

```diff
@@ -452,5 +547,358 @@ <h1>
+        1. Return CreateArrayFromList(_sorted_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-hourcyclesoflocale" type="abstract operation">
+      <h1>
+        HourCyclesOfLocale (
+          _loc_: an Intl.Locale,
+        ): an Array
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>The following algorithm refers to Locale data specified in <a href="https://unicode.org/reports/tr35/tr35-dates.html#Time_Data">UTS 35's Time Data</a>.</dd>
+      </dl>
+      <emu-alg>
+        1. If _loc_.[[HourCycle]] is not *undefined*, then
+          1. Return CreateArrayFromList(&laquo; _loc_.[[HourCycle]] &raquo;).
+        1. Let _preference_ be RegionPreference(_loc_.[[Locale]]).
+        1. Let _region_ be _preference_.[[Region]].
+        1. Let _regionOverride_ be _preference_.[[RegionOverride]].
+        1. If _regionOverride_ is not *undefined* and time data for _regionOverride_ are available, then
+          1. Let _lookupRegion_ be _regionOverride_.
+        1. Else,
+          1. Let _lookupRegion_ be _region_.
+        1. Let _list_ be a List of unique hour cycle identifiers, which must be lower case String values indicating either the 12-hour format (*"h11"*, *"h12"*) or the 24-hour format (*"h23"*, *"h24"*), sorted in descending preference of those in common use for date and time formatting in _lookupRegion_. The list is empty if no time data for _lookupRegion_ is available.
+        1. If _list_ is empty, set _list_ to &laquo; *"h23"* &raquo;.
+        1. Return CreateArrayFromList(_list_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-numberingsystemsoflocale" type="abstract operation">
+      <h1>
+        NumberingSystemsOfLocale (
+          _loc_: an Intl.Locale,
+        ): an Array
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. If _loc_.[[NumberingSystem]] is not *undefined*, then
+          1. Return CreateArrayFromList(&laquo; _loc_.[[NumberingSystem]] &raquo;).
+        1. Let _r_ be LookupMatchingLocaleByPrefix(%Intl.NumberFormat%.[[AvailableLocales]], &laquo; _loc_.[[Locale]] &raquo;).
+        1. If _r_ is not *undefined*, then
+          1. Let _foundLocale_ be _r_.[[locale]].
+          1. Let _foundLocaleData_ be %Intl.NumberFormat%.[[LocaleData]].[[&lt;_foundLocale_&gt;]].
+          1. Let _numberingSystems_ be _foundLocaleData_.[[nu]].
+          1. Let _list_ be &laquo; _numberingSystems_[0] &raquo;.
+        1. Else,
+          1. Let _list_ be &laquo; *"latn"* &raquo;.
+        1. Return CreateArrayFromList(_list_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-timezonesoflocale" type="abstract operation">
+      <h1>
+        TimeZonesOfLocale (
+          _loc_: an Intl.Locale,
+        ): an Array
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _region_ be GetLocaleRegion(_loc_.[[Locale]]).
+        1. Assert: _region_ is not *undefined*.
```

Comment:
This assertion is unjustified.
```suggestion
        TimeZonesOfLocale (
          _loc_: an Intl.Locale,
        ): an Array or *undefined*
      </h1>
      <dl class="header">
      </dl>
      <emu-alg>
        1. Let _region_ be GetLocaleRegion(_loc_.[[Locale]]).
        1. If _region_ is *undefined*, return *undefined*.
```